### PR TITLE
Fix left multiplying operators with numpy arrays

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-58675]

**Context:**

`np.array([0.5, 0.6]) * qml.S(0)` should give a `SProd` op with a batch dimension.

**Description of the Change:**

Setting the `__array_priority__` property tells numpy we should use `Operator.__mul__` and `Operator.__rmul__` instead of the corresponding numpy methods that try and stick an operator into an array.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by Qottmann — APPROVED** [review]
nice!

**Review by lillian542 — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -60,6 +60,8 @@
 
 <h3>Improvements 🛠</h3>
 
+* Operators can now be left multiplied `x * op` by numpy arrays.
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:63]: missing link to PR here

Comment (albi3ro) [reply doc/releases/changelog-dev.md:63]:
```suggestion
* Operators can now be left multiplied `x * op` by numpy arrays.
  [(#5361)](https://github.com/PennyLaneAI/pennylane/pull/5361)
```

File: tests/test_operation.py [hunk 2]
```diff
@@ -1197,6 +1197,24 @@ def test_mul_scalar_tensor(self):
         assert isinstance(prod_op, SProd)
         assert prod_op.scalar is scalar
 
+        prod_op2 = scalar * qml.RX(1.23, 0)
+        assert isinstance(prod_op2, SProd)
+        assert prod_op.scalar is scalar
+
+    def test_mul_array_nupy(self):
```
Comment (albi3ro) [inline tests/test_operation.py:1204]:
```suggestion
    def test_mul_array_numpy(self):
```

**Pull request merged**